### ISSUE TITLE
Intermittent linux SSL issue

### ISSUE THREAD

**Issue by robchallen** [bristol-vaccine-centre/rsurvstat#2]
SSL problem: 7.81.0
OpenSSL/3.0.21.2.11
libssh/0.9.6/openssl/zlib2.3.2x86_64-pc-linux-gnu
  ! error:0A000126:SSL routines::unexpected eof while reading

Doesn;t seem to be very consistent and happens intermittently on all platforms.

**Comment by robchallen** [comment]
as of openSSL 3.0 this maybe somethign to do with SSL_OP_IGNORE_UNEXPECTED_EOF but I can;t make it work.

**Comment by robchallen** [comment]
Currently, in Ubuntu 22.04 LTS curl is version 7.81.0, and fails in case of unexpected EOF with OpenSSL 3.0.x. The latest curl is 7.88.1, and apparently resolved the issue, though I could not find the exact commit or ticket. The solution in Ubuntu 22.04 and any system with old curl is to manually compile and install the latest one, as shown for example [here](https://yannmjl.medium.com/how-to-manually-update-curl-on-ubuntu-server-899476062ad6).

**Comment by robchallen** [comment]
TLDR:

apt remove curl
apt purge curl
apt-get update
apt-get install -y libssl-dev autoconf libtool make
cd /usr/local/src
wget https://curl.haxx.se/download/curl-7.88.1.zip
unzip curl-7.88.1.zip
cd curl-7.88.1
./buildconf
./configure --with-ssl 
make
sudo make install
sudo cp /usr/local/bin/curl /usr/bin/curl
sudo ld

**Comment by robchallen** [comment]
Windows github runners seem to sometimes fail with 

SSL problem: 
7.84.0(OpenSSL/1.1.1m) 
Schannel1.2.11
libssh2/1.10.1

  ! schannel: failed to receive handshake, SSL/TLS connection failed
